debian/rules

```diff
@@ -1,18 +1,51 @@
 #!/usr/bin/make -f
+#
+# Build Debian package using https://github.com/spotify/dh-virtualenv
+#
+# The below targets create a clean copy of the workdir via
+# using "sdist", else "pip" goes haywire when installing from
+# sourcedir ".", because that includes the debian build stage,
+# and a recursive explosion ensues when symlinks are followed.
+#
+# It also ensures your MANIFEST is complete and at least covers
+# all files needed for a release build.
 
-# This file was automatically generated by stdeb 0.8.2 at
-# Fri, 12 Jun 2015 14:32:03 +0100
-export PYBUILD_NAME=matrix-synapse
-%:
-	dh $@ --with python2 --with systemd --buildsystem=pybuild --no-guessing-deps
+# Increase trace logging, see debhelper(7) (uncomment to enable)
+#DH_VERBOSE=1
+
+export DH_VIRTUALENV_INSTALL_ROOT=/opt/venvs
+export CFLAGS=-march=x86-64
+SNAKE=/usr/bin/python3
+DEB_BUILD_OPTIONS=noddebs
+EXTRA_REQUIREMENTS=--preinstall "setuptools>=17.1" --preinstall "pip>=7" --preinstall "wheel" --preinstall "lxml"
+DH_VENV_ARGS=--install-suffix "matrix-synapse" --with python-virtualenv --setuptools --python $(SNAKE) $(EXTRA_REQUIREMENTS) --extra-pip-arg="--no-cache-dir" --extra-pip-arg="--compile"
+PACKAGE=$(shell dh_listpackages)
+VERSION=$(shell $(SNAKE) setup.py --version)
+SDIST_DIR=debian/$(PACKAGE)-$(VERSION)
+RELEASE_VERSION=$(VERSION)~$(shell lsb_release -c -s)
```

Comment:
do we really need separate version numbers for different dists? it doesn't sound like a very debian thing to do to me.

Reply:
(I'm also wondering if `~` is the right separator here, though I guess it is as good as any other)

Reply:
It's what I've seen before, and we do need to make different debs for different diffs (because targeting different Python 3 versions). https://www.debian.org/doc/debian-policy/ch-controlfields.html#s-f-version says it looks fine, I think

Reply:
ok, seems fair.